### PR TITLE
bugfix/ST-403-duplicated-songbirdjs

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/core/integrations/CardinalCommerce.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ export class CardinalCommerce {
   }
 
   protected _threeDSetup() {
-    if (!this._called) {
+    if (!this._called && !DomMethods.isScriptLoaded(this._sdkAddress)) {
```
Comment (james-simpson-ST) [inline src/core/integrations/CardinalCommerce.ts]: Should it be the responsibility of the insertScript to check isScriptLoaded? That way we don't need to change both CardinalCommerce and VisaCheckout. If it isn't done this way don't we need to change VisaCheckout?

Also do we still need to call _onCardinalLoad if we have a fresh instance of CardinalCommerce - currently I don't think we will but I just wanted to check we don't want to?

File: src/json/config.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
       "color-error": "#3358FF"
     }
   },
-  "submitOnSuccess": false,
+  "submitOnSuccess": true,
```
Comment (james-simpson-ST) [inline src/json/config.json]: Does this want to be reverted before we merge back?

**Comment by james-simpson-ST** [comment]
Only just noticed this is a WIP - but I've done the review now anyway 😉

File: src/core/integrations/CardinalCommerce.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/core/integrations/CardinalCommerce.ts]: Ad.1. Yes, your're right I'll fix this.
Ad.2. Not sure, bu probably this is not needed; I'll take a look at this.

File: src/json/config.json (hunk 2 above)
Comment (pgs-gmartowski) [reply src/json/config.json]: yup, thx, this is one of my `debugging` changes ;)

File: src/core/integrations/CardinalCommerce.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/core/integrations/CardinalCommerce.ts]: Ad.1. Done,
Ad.2. Seems to work fine.

File: src/json/config.json (hunk 2 above)
Comment (pgs-gmartowski) [reply src/json/config.json]: done

File: src/core/shared/DomMethods.ts [hunk 3]
```diff
@@ -1,106 +1,118 @@
...
+    return script;
   }
 
-  public static addListener(targetId: string, listenerType: string, callback: any) {
-    document.getElementById(targetId).addEventListener(listenerType, callback);
+  public static insertStyle(contents: string): void {
+    const head = document.getElementById('insertedStyles');
+    if (head) {
+      return;
+    }
```
Comment (pgs-gmartowski) [inline src/core/shared/DomMethods.ts]: info: this change has been added because of duplicating styles in head whih was inserted by updateJwt function

**Pull request merged**